examples/tpm_io.c

```diff
@@ -211,6 +221,10 @@
         int mode = 0; /* Mode 0 (CPOL=0, CPHA=0) */
         int bits_per_word = 8; /* 8-bits */
 
+    #ifdef WOLFTPM_AUTODETECT
+    tryagain:
```

Comment:
This label & later goto make sense. Just to be thorough, what about having the count of supported modules (right now 4) and having a simpler for/while cycle going through detection?
My consideration is only one - With time, the number of supported modules will rise and having that list as an actual array(programming list) to go through might be useful in future.
So, this is more of an enhancement rather than request for change. Because the goto here makes sense, keeps things simple, changes CS0 and CS1, can detect everything and anything.